interface/billing/billing_process.php

```diff
@@ -515,19 +49,17 @@ function process_form($ar)
             <div class="col-12">
                 <h3><?php echo xlt('Billing queue results'); ?>:</h3>
                 <ul>
-                    <li>
-                        <?php
-                        foreach ($bill_info as $infoline) {
-                            echo nl2br($infoline);
-                        }
-                        ?>
-                    </li>
+                    <?php  foreach ($logger->bill_info() as $infoline) { ?>
+                        <li>
+                            <?php echo nl2br($infoline); ?>
```

Comment:
does this $infoline variable ever contain html elements?
If not, it should be escaped:
```
<?php echo nl2br(text($infoline)); ?>
```

Reply:
This is how it was, and it doesn't currently, but I suppose another developer could print HTML to the log.

Reply:
then lets add the text() to it, which will then ensure html escaping here